### PR TITLE
Kind: add an option to create a second interface in all nodes

### PR DESCRIPTION
**Pull request by fedepaol**
**- What this PR does and why is it needed**

When running kind with -eb, we create a second network and add all the
node-containers to it to create a second interface into "nodes".
Then, we use the value of the same parameter to instruct ovnkube that we
intend to use a different nic for external gateway traffic.

**- Special notes for reviewers**

This is based to the additional parameters introduced in https://github.com/ovn-org/ovn-kubernetes/pull/2270

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: contrib/kind.sh [hunk 1]
```diff
@@ -566,6 +574,21 @@ cleanup_kube_proxy_iptables() {
   done
 }
 
+docker_create_second_interface() {
+  if [ -z "$OVN_EX_GW_NETWORK_INTERFACE" ]; then
+    return
+  fi
```
Comment (aojea) [inline contrib/kind.sh:580]: I always prefer to do the if from the main loop

File: contrib/kind.sh [hunk 2]
```diff
@@ -583,6 +606,7 @@ check_ipv6
 set_cluster_cidr_ip_families
 create_kind_cluster
 docker_disable_ipv6
+docker_create_second_interface
```
Comment (aojea) [inline contrib/kind.sh:609]: my bash is not great checks if that is correct
```suggestion
  if [ ! -z "$OVN_EX_GW_NETWORK_INTERFACE" ]; then
    docker_create_second_interface
  fi
```

File: contrib/kind.sh (hunk 1 above)
Comment (aojea) [reply contrib/kind.sh:580]: so you don't have to go down into the functions to understand what happens

Comment (fedepaol) [reply contrib/kind.sh:580]: Good point! That improves readability a lot

Comment (fedepaol) [reply contrib/kind.sh]: Done

File: contrib/kind.sh [hunk 3]
```diff
@@ -164,6 +166,9 @@ parse_args() {
                                                 fi
                                                 DBCHECKER_LOG_LEVEL=$1
                                                 ;;
+            -eb | --egress-bridge )               shift
```
Comment (trozet) [inline contrib/kind.sh:169]: nit extra spaces on the shift

File: contrib/kind.sh [hunk 4]
```diff
@@ -544,6 +552,18 @@ kubectl_wait_pods() {
   fi
 }
 
+docker_create_second_interface() {
+  echo "adding second interfaces to nodes"
+
+  # Create the network as dual stack, regardless of the type of the deployment. Ignore if already exists.
+  docker network create --ipv6 --driver=bridge network2 --subnet=172.19.0.0/16 --subnet=fc00:f853:ccd:e798::/64 || true
```
Comment (trozet) [inline contrib/kind.sh:559]: should this network be called kind2 or something?

File: dist/images/ovnkube.sh [hunk 5]
```diff
@@ -934,6 +942,7 @@ ovn-master() {
     ${egressip_enabled_flag} \
     ${egressfirewall_enabled_flag} \
     --metrics-bind-address ${ovnkube_master_metrics_bind_address} \
+    ${egress_interface} \
```
Comment (trozet) [inline dist/images/ovnkube.sh:945]: why do we need it on the master?

File: contrib/kind.sh (hunk 4 above)
Comment (fedepaol) [reply contrib/kind.sh:559]: I wanted to make it clear it's not related to the primary network, thought that something different from kind* was better.
Can change to "exgwnetwork" or something along that line though

File: dist/images/ovnkube.sh (hunk 5 above)
Comment (fedepaol) [reply dist/images/ovnkube.sh:945]: That's right, we can do everything with the node parameter. I'll change that in the other pr

File: contrib/client.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (trozet) [inline contrib/client.yaml:1]: accident?

**Review by trozet — CHANGES_REQUESTED** [review]

File: contrib/kind.sh [hunk 7]
```diff
@@ -169,6 +171,9 @@ parse_args() {
                                                 fi
                                                 DBCHECKER_LOG_LEVEL=$1
                                                 ;;
+            -eb | --egress-bridge )             shift
```
Comment (trozet) [inline contrib/kind.sh:174]: this doesn't match the usage:
```
--egress-gw-bridge
```

**Review by trozet — CHANGES_REQUESTED** [review]

File: contrib/kind.sh (hunk 7 above)
Comment (fedepaol) [reply contrib/kind.sh]: Done

File: contrib/kind.sh [hunk 8]
```diff
@@ -77,6 +78,7 @@ usage() {
     echo "-cl  | --ovn-loglevel-controller   Log config for ovn-controller DEFAULT: '-vconsole:info'."
     echo "-dl  | --ovn-loglevel-nbctld       Log config for nbctl daemon DEFAULT: '-vconsole:info'."
     echo "-ep  | --experimental-provider     Use an experimental OCI provider such as podman, instead of docker. DEFAULT: Disabled."
+    echo "-eb  | --egress-gw-bridge          The interface to be used as bridge for external gw traffic."
```
Comment (trozet) [inline contrib/kind.sh:81]: why even give the option here of providing an interface name? We know kind deployments have eth0 as their first interface, isn't it predictable what the 2nd interface will be? Also, when you connect the docker containers to the second docker network, there is no way to name the interface they get in order to match what is passed in via this arg right?

Seems like they could just provide the argument, and the description could be something like "Uses an alternate interface for egress gateway traffic".

Comment (fedepaol) [reply contrib/kind.sh:81]: ah yes, makes sense. The second interface is always eth1 , that's what we pass in CI in https://github.com/ovn-org/ovn-kubernetes/pull/2289/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R355 

Did not think about abstracting away the interface name, I'll make that a flag.

**Review by trozet — APPROVED** [review]
/lgtm

**Pull request merged**